rholang/src/main/scala/coop/rchain/rholang/interpreter/matcher/package.scala

```diff
@@ -24,37 +28,38 @@ package object matcher {
   type NonDetFreeMapWithCost[A] = StateT[StreamWithCost, FreeMap, A]
   type StreamWithCost[A]        = StreamT[ErroredOrCostA, A]
 
+  type _freeMap[F[_]] = MonadState[F, FreeMap]
```

Comment:
Why the `_` prefixes? Are you trying to express something which should be expressed with an access modifier?

Reply:
This convention is inspired by the [eff library](http://atnos-org.github.io/eff/org.atnos.site.Tutorial.html#write-an-interpreter-for-your-program) and I use it because it elegantly avoids the name conflict between the `Cost` typeclass and the `_cost` effect. I could go with suffixes, but it'd:
 1. be more verbose
 2. expose implementation detail that's likely to change (_cost is likely to be writer in a while, and _error to be ApplicativeError/ApplicativeThrow vs MonadError)

In my mind, this convention means: this is an effect-type alias. Will be used similarly to capabilities, but for more generic and probably low-level/implementation stuff.

Reply:
Makes sense. Shouldn't it read `_freeMapState` then?

Reply:
I'd like to avoid the suffix for the same reason `_error` is not `_monadError` :)